### PR TITLE
 #403 Handle batch_presence_aware

### PR DESCRIPTION
**Pull request by suekto-andreas**
This is in relation to issue #403 

In order to support this:
1. batch_presence_aware is treated as a connection parameter
2. introduce a connParams map to host parameter to embed into the generated connection URL
3. example of usage is given in example/connparams/connparams.go

Kindly help to review, please feel free to feedback. Thanks !

### CODE REVIEW

File: examples/connparams/connparams.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+func main() {
+	api := slack.New(
+		"YOUR TOKEN HERE",
+		slack.OptionDebug(true),
+		slack.OptionLog(log.New(os.Stdout, "slack-bot: ", log.Lshortfile|log.LstdFlags)),
+	)
+
+	// turn on the batch_presence_aware option
+	rtm := api.NewRTM(slack.RTMOptionConnParams(map[string]string{
```
Comment (james-lawrence) [inline examples/connparams/connparams.go]: I believe you want [url.Values](https://golang.org/pkg/net/url/#Values) here.

Comment (suekto-andreas) [reply examples/connparams/connparams.go]: Thanks! yep that express it better - updated.

**Comment by james-lawrence** [comment]
LGTM, tyvm.

**Comment by ramikalai** [comment]
Does this actually work? The docs seem to say that this param needs to be passed to rtm.connect (this is the function which returns the websocket URL)... I don't believe this is the case currently and Im not receiving any presence events

**Comment by suekto-andreas** [comment]
@ramikalai  I have been not using this for a while, unless there is Slack API changes, it was working. Could you help to share how to replicate your finding ? (i.e. snippet of code etc.)

Probably a bug ticket is a better approach to communicate, complete with replication step.

**Comment by ramikalai** [comment]
Hey @suekto-andreas, I was doing something silly! Got it to work using the current implementation. Thank you!

**Pull request merged**